pkg/storage/engine/mvcc.go

```diff
@@ -1852,6 +1852,23 @@ func mvccResolveWriteIntent(
 	// For cases where there's no write intent to resolve, or one exists
 	// which we can't resolve, this is a noop.
 	if !ok || meta.Txn == nil || !roachpb.TxnIDEqual(intent.Txn.ID, meta.Txn.ID) {
```

Comment:
Is this new code applicable in all three of these cases? I'm not exactly clear on what each of these conditions represents, but my guesses:
`!ok`: there's no value here at all
`meta.Txn == nil`: there's a value, but no intent
`!roachpb.TxnIDEqual(intent.Txn.ID, meta.Txn.ID)`: there's an intent, but it belongs to another transaction

Does `!ok` necessary mean the `mvccGetUsingIter` will return an error?